pretix_eth/payment.py

```diff
@@ -223,7 +226,24 @@ def payment_control_render(self, request: HttpRequest, payment: OrderPayment):
     abort_pending_allowed = True
 
     def payment_refund_supported(self, payment: OrderPayment):
-        return False
+        return payment.state == OrderPayment.PAYMENT_STATE_CONFIRMED
 
     def payment_partial_refund_supported(self, payment: OrderPayment):
-        return False
+        return self.payment_refund_supported(payment)
+
+    def execute_refund(self, refund: OrderRefund):
+        if refund.payment is None:
+            raise Exception('Invariant: No payment associated with refund')
+
+        wallet_queryset = WalletAddress.objects.filter(order_payment=refund.payment)
+
+        if wallet_queryset.count() != 1:
+            raise ValueError('There is not assigned wallet address to this payment')
```

Comment:
I understand your concern. However, looks like that it cannot be implemented from the plugin part and should be the concern for pretix. 

At the same time, I do not see this happening for the user. The test is very artificial - it assumes that payment was already confirmed, but without actually having an associated wallet address (which is impossible). 

It is possible that a request for a refund happened at the moment between the payment object created and the wallet address about to be assigned. However, the request will first check if that payment supports automatic refunds, in our case, it will first verify if `payment.state == OrderPayment.PAYMENT_STATE_CONFIRMED`. Considering that the current payment is not at this state, the automatic refund request will be rejected and become manual instead. 

I actually should remove this test because it is quite confusing and change the exception from `ValueError` to `Exception`.